### PR TITLE
archive all cards from stack

### PR DESCRIPTION
**Pull request by jakobroehrl**
Signed-off-by: Jakob Röhrl <jakob.roehrl@web.de>
* Resolves: #1711
* Target version: master 

grafik

### CODE REVIEW

**Comment by juliushaertl** [comment]
@jakobroehrl Enhancements should go to the next minor release so this will need to wait for 1.1.0. :wink:

File: src/components/board/Stack.vue [hunk 1]
```diff
@@ -178,6 +181,11 @@ export default {
 		deleteStack(stack) {
 			this.$store.dispatch('deleteStack', stack)
 		},
+		archiveAllCardsFromStack(stack) {
+			this.cardsByStack.forEach(card => {
```
Comment (juliushaertl) [inline src/components/board/Stack.vue:185]: Not fully convinced about this, we should somehow have a progress shown about this and make sure that the changes are not done until all the archive operations have been performed.

**Comment by stefan-niedermann** [comment]
I would switch the entries, like in the boards menu:

grafik

In general we should put the most destructive operations (delete in this case) always at the bottom (Files app does this, too).

**Comment by jakobroehrl** [comment]
> In general we should put the most destructive operations (delete in this case) always at the bottom (Files app does this, too).

done

File: src/components/board/Stack.vue [hunk 2]
```diff
@@ -123,6 +140,8 @@ export default {
 			showAddCard: false,
 			stateCardCreating: false,
 			animate: false,
+			modalShow: false,
+			archiveAllCardsProgress: '',
```
Comment (juliushaertl) [inline src/components/board/Stack.vue:144]: Please use a sane default like `null`

File: src/components/board/Stack.vue [hunk 3]
```diff
@@ -51,6 +54,19 @@
 			</Actions>
 		</div>
 
+		<Modal v-if="modalShow" title="Archive all cards in this list" @close="modalShow=false">
+			<div class="modal__content">
+				<h3>Archive all cards in this list</h3>
+				<progress :value="archiveAllCardsProgress" :max="cardsByStack.length" />
```
Comment (juliushaertl) [inline src/components/board/Stack.vue:60]: cardsByStack will decrease during the process, so ideally you store the max in a local data field before starting the process

File: src/components/board/Stack.vue [hunk 4]
```diff
@@ -51,6 +54,19 @@
 			</Actions>
 		</div>
 
+		<Modal v-if="modalShow" title="Archive all cards in this list" @close="modalShow=false">
+			<div class="modal__content">
+				<h3>Archive all cards in this list</h3>
```
Comment (juliushaertl) [inline src/components/board/Stack.vue:59]: Should be translated

File: src/components/board/Stack.vue [hunk 5]
```diff
@@ -51,6 +54,19 @@
 			</Actions>
 		</div>
 
+		<Modal v-if="modalShow" title="Archive all cards in this list" @close="modalShow=false">
```
Comment (juliushaertl) [inline src/components/board/Stack.vue:57]: No need for a title if we have a header in the modal

File: src/components/board/Stack.vue [hunk 6]
```diff
@@ -123,6 +140,9 @@ export default {
 			showAddCard: false,
 			stateCardCreating: false,
 			animate: false,
+			modalShow: false,
```
Comment (juliushaertl) [inline src/components/board/Stack.vue:143]: Please give it a more obvious name on what kind of modal this is

File: src/components/board/Stack.vue [hunk 7]
```diff
@@ -123,6 +140,9 @@ export default {
...
+			archiveAllCardsProgress: null,
+			stackLen: 0,
```
Comment (juliushaertl) [inline src/components/board/Stack.vue:145]: lets group those into an object:

```
stackTransfer: {
  total: 0,
  current: null,
}
```

File: src/components/board/Stack.vue [hunk 8]
```diff
@@ -299,4 +328,25 @@ export default {
...
 	}
 
+	.modal__content {
+		width: 25vw;
+		min-width: 250px;
+		min-height: 100px;
+		text-align: center;
+		margin: 20px 20px 20px 20px;
+
+		.multiselect {
```
Comment (juliushaertl) [inline src/components/board/Stack.vue:338]: Why?

Comment (juliushaertl) [reply src/components/board/Stack.vue:355]: This still needs to be removed, there is no multiselect in here.

File: src/components/board/Stack.vue (hunk 7 above)
Comment (juliushaertl) [reply src/components/board/Stack.vue:150]: @jakobroehrl What about this?

**Pull request merged**